### PR TITLE
Apply camera

### PR DESCRIPTION
**Pull request by TimYuenior**
Fix the preset loading mainly in the director UI.

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/gui/DirectorLiveController.java [hunk 1]
```diff
@@ -103,6 +99,16 @@
...
         
+        if (!script.isEmpty()) {
+            if (script.getCurrent() > -1) {
+                btnNext.fire();
+            } else {
+                script.initPresetLoading();
+            }
+        }
+
+        //bindImageToBox(thumbnail, thumbnailBox);
```
Comment (tkolenbrander) [inline src/main/java/nl/tudelft/contextproject/gui/DirectorLiveController.java]: What is this commented code and why is it commented?

File: src/test/java/nl/tudelft/contextproject/script/TimelineTest.java [hunk 2]
```diff
@@ -100,6 +100,25 @@ public void testNextPreset() {
...
+        timeline1.instantNextPreset();
+        assertEquals(cam0.getSettings(), new CameraSettings(1, 1, 1, 2));
+        assertEquals(timeline1.getCurrent(), 0);
+        cam0.tilt(20);
+        assertEquals(timeline1.getCurrent(), 0);
+        timeline1.instantNextPreset();
+        System.out.println(cam0.getSettings().getPan() + " " + cam0.getSettings().getTilt() + " "
+                + cam0.getSettings().getZoom() + " " + cam0.getSettings().getFocus());
+        System.out.println(pres.getToSet().getPan() + " " + pres.getToSet().getTilt() + " "
+                + pres.getToSet().getZoom() + " " + pres.getToSet().getFocus());
```
Comment (tkolenbrander) [inline src/test/java/nl/tudelft/contextproject/script/TimelineTest.java]: Why do you have prints in your test?

File: src/main/java/nl/tudelft/contextproject/view/DirectorLiveView.fxml [hunk 3]
```diff
@@ -31,9 +31,15 @@
...
+      <CheckBox fx:id="automaticCheck" layoutX="348.0" layoutY="534.0" mnemonicParsing="false" selected="true" text="Automatic preset loading" AnchorPane.bottomAnchor="20.0" AnchorPane.rightAnchor="256.0" />
+=======
       <Label fx:id="actionTxt" layoutX="203.0" layoutY="263.0" prefHeight="17.0" prefWidth="190.0" textFill="RED" AnchorPane.bottomAnchor="295.0" AnchorPane.leftAnchor="203.0" />
       <Button fx:id="btnNext" contentDisplay="CENTER" layoutX="701.0" layoutY="532.0" maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" mnemonicParsing="false" prefHeight="31.0" prefWidth="84.0" text="Go live" AnchorPane.bottomAnchor="15.0" AnchorPane.rightAnchor="15.0" />
       <CheckBox fx:id="automaticCheck" layoutX="553.0" layoutY="536.0" mnemonicParsing="false" selected="true" text="Automatic loading" AnchorPane.bottomAnchor="22.0" AnchorPane.rightAnchor="128.0" />
+>>>>>>> fixez
```
Comment (tkolenbrander) [inline src/main/java/nl/tudelft/contextproject/view/DirectorLiveView.fxml]: This is still an unresolved merge conflict.

File: src/main/java/nl/tudelft/contextproject/gui/DirectorLiveController.java [hunk 4]
```diff
@@ -118,11 +121,25 @@ private void initializeButtons() {
...
+                script.next();
+            }
+            tableShots.refresh();
             initializeLive();
             btnNext.setText("Next shot");
         });
+        
+        btnManualLoad.setOnAction(event -> {
+            script.loadNextPresets();
+            System.out.println("dkdkddk");
```
Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/gui/DirectorLiveController.java]: Please remove this `println`.

Comment (Ettatabe) [inline src/main/java/nl/tudelft/contextproject/gui/DirectorLiveController.java]: delete this system.out.println

File: src/main/java/nl/tudelft/contextproject/script/Script.java [hunk 5]
```diff
@@ -279,19 +290,19 @@ public int hashCode() {
      */
     @Override
     public Shot next() {
-        return next(false);
+        return next(true);
     }
     
     /**
      * Go to the next shot.
      * Depending on boolean skip, cameras are adjusted or not.
```
Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/script/Script.java:298]: `boolean skip` is the old name of the variable. Please update this reference

File: src/main/java/nl/tudelft/contextproject/script/Timeline.java [hunk 6]
```diff
@@ -80,12 +82,35 @@ public void initPreset() {
      * @param oldShot The shot that just finished.
      */
     public void nextPreset(Shot oldShot) {
-        int oldIndex = shots.indexOf(oldShot);
+        
+        int oldIndex;
+        
+        if (oldShot == null) {
```
Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/script/Timeline.java]: This kind of if-statement is exactly what advanced if-statements were made for.

File: src/test/java/nl/tudelft/contextproject/script/TimelineTest.java [hunk 7]
```diff
@@ -100,6 +100,21 @@ public void testNextPreset() {
...
+    @Test
+    public void testInstantNextPreset() {
+        timeline1.instantNextPreset();
+        assertEquals(cam0.getSettings(), new CameraSettings(1, 1, 1, 2));
+        assertEquals(timeline1.getCurrent(), 0);
+        cam0.tilt(20);
+        assertEquals(timeline1.getCurrent(), 0);
+        timeline1.instantNextPreset();
+        assertEquals(cam0.getSettings(), new CameraSettings(1, 1, 1, 2));
+
```
Comment (bvobart) [inline src/test/java/nl/tudelft/contextproject/script/TimelineTest.java]: Remove this whitespace please

**Comment by bvobart** [comment]
I've placed some comments. Once these are fixed, then this branch can be merged.

**Pull request merged**